scipy/linalg/tests/test_lapack.py

```diff
@@ -1645,3 +1645,158 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+
+def test_gttrf_gttrs():
+    #%%
+    np.random.seed(42)
+    # use ?gttrf and ?gttrs to solve a random linear system for each dtype
+    for index, dtype in enumerate(DTYPES):
+        n = 10
+        # set appropriate test tolerance based on the data type
+        if index % 2 == 0:
```

Comment:
Does this need to depend on the data type? I would prefer to avoid the `if` statement.

Reply:
Again, `tol=25*np.finfo(dtype).eps` appears elsewhere and it works fine here. And I think you can just use this number for both `rtol` and `atol`.

Reply:
I hadn't used `rtol=25*np.finfo(dtype).eps` here since it is a little too small to pass over the tests. For example, when `rtol` is `rtol=2.22045e-14`, an error is raised due to a max relative difference of `5.25248037e-14`. 

Right now I'll use `rtol=250*np.finfo(dtype).eps` so that `rtol` is` 5.5511...e-14`, unless you have another preference.

Reply:
I think they passed on my computer, so I thought it was fine. But an extra factor of 10 is fine, too.